core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -448,6 +473,20 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
    */
   private[spark] def getenv(name: String): String = System.getenv(name)
 
+  /**
+   * Wrapper method for get*() methods which require some specific value format. This catches
+   * any [[NumberFormatException]] or [[IllegalArgumentException]] and re-raises it with the
+   * incorrectly configured key in the exception message.
+   */
+  private def catchIllegalArgument[T](key: String)(getValue: => T): T = {
```

Comment:
According to what it actually does `catchIllegalArgument` don't seems to be a great name for this function, maybe `catchIllegalValue`?